deployment/deployment.go

```diff
@@ -0,0 +1,85 @@
+package deployment
+
+import (
+	"os"
+
+	"github.com/akitasoftware/akita-cli/printer"
+	"github.com/akitasoftware/akita-libs/tags"
+)
+
+type Deployment string
+
+const (
+	None       Deployment = ""
+	Unknown               = "unknown"
+	Kubernetes            = "kubernetes"
+)
+
+func (d Deployment) String() string {
+	return string(d)
+}
+
+func GetDeploymentInfo() (Deployment, map[tags.Key]string) {
+	// Allow the user to specify a deployment environment, even
+	// if it's of an unknown type.
+
+	deploymentType := None
+	tagset := make(map[tags.Key]string)
+
+	if d := os.Getenv("AKITA_DEPLOYMENT"); d != "" {
+		deploymentType = Unknown
+		tagset[tags.XAkitaDeployment] = d
+	}
+
+	// If there is a git commit associated with this deployment,
+	// then record it.
+	if c := os.Getenv("AKITA_DEPLOYMENT_COMMIT"); c != "" {
+		tagset[tags.XAkitaGitCommit] = c
+	}
+
+	if GetAWSTags(tagset) {
+		printer.Infof("Found AWS environment variables.\n")
+	}
+
+	if GetKubernetesTags(tagset) {
+		printer.Infof("Found Kubernetes environment variables.\n")
+		deploymentType = Kubernetes
+	}
+
+	return deploymentType, tagset
+}
+
+func GetKubernetesTags(tagset map[tags.Key]string) bool {
+	// Return true if any of these are present.
+	envMapping := []struct {
+		EnvVar string
+		Key    tags.Key
+	}{
+		{"AKITA_K8S_NAMESPACE", tags.XAkitaKubernetesNamespace},
+		{"AKITA_K8S_NODE", tags.XAkitaKubernetesNode},
+		{"AKITA_K8S_HOST_IP", tags.XAkitaKubernetesHostIP},
+		{"AKITA_K8S_POD", tags.XAkitaKubernetesPod},
+		{"AKITA_K8S_POD_IP", tags.XAkitaKubernetesPodIP},
+		{"AKITA_K8S_DAEMONSET", tags.XAkitaKubernetesDaemonset},
+	}
```

Comment:
Would be nice if we could have the environment–tag mapping in one place. Right now, it's spread across three functions: here, in `GetDeploymentInfo`, and in `GetAWSTags`.

What if we had a two-level map at the package level? Something like:
```
type mappingGroup int
const (
	defaultGroup = iota
	awsGroup
	kubernetesGroup
)

// Each value maps a tag key to its corresponding environment variable.
var envMapping = map[mappingGroup]map[tags.Key]string{
	defaultGroup: {
		tags.XAkitaDeployment: "AKITA_DEPLOYMENT",
		tags.XAkitaGitCommit: "AKITA_DEPLOYMENT_COMMIT",
	},
	awsGroup: {
		tags.XAkitaAWSRegion: "AKITA_AWS_REGION",
	},
	kubernetesGroup: {
		tags.XAkitaKubernetesNamespace: "AKITA_K8S_NAMESPACE",
		...
	},
}
```
Then, the job of extracting tags from the environment can be factored into something that looks like `GetKubernetesTags`, but also called from the other two places.

Reply:
Good idea, implemented.